pandas/core/groupby.py

```diff
@@ -465,12 +466,21 @@ def _set_grouper(self, obj, sort=False):
             raise ValueError(
                 "The Grouper cannot specify both a key and a level!")
 
+        # Keep self.grouper value before overriding
+        if self._grouper is None:
+            self._grouper = self.grouper
+
         # the key must be a valid info item
         if self.key is not None:
             key = self.key
-            if key not in obj._info_axis:
-                raise KeyError("The grouper name {0} is not found".format(key))
-            ax = Index(obj[key], name=key)
+            if getattr(self.grouper, 'name', None) == key and \
```

Comment:
can you add a comment here about what this is doing (IOW the on is already defined).

use ``ABCSeries`` rather than ``Series``